### PR TITLE
Add notifications for expired tests

### PR DESCRIPTION
**Pull request by maxim12321**

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/converters/notifications/TestExpiredNotificationConverter.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public NotificationDTO convertToDTO(Notification notification) {
+        return notificationBuilder(notification)
+                .level(notification.getTest().getLevel().getName())
+                .build();
+    }
+
+    @Override
+    public NotificationType converterType() {
+        return NotificationType.TEST_EXPIRED;
+    }
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/converters/notifications/TestExpiredNotificationConverter.java:20]: I think, we can pretty easy make the same with level

File: src/main/java/com/team4/testingsystem/entities/Timer.java [hunk 2]
```diff
@@ -25,9 +31,16 @@
...
+    private Status status;
+
+    @Column(name = "scheduled_time")
+    private Instant scheduledTime;
+
+    public Timer(Test test, Status status, Instant scheduledTime) {
         this.test = test;
+        this.status = status;
+        this.scheduledTime = scheduledTime;
     }
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/entities/Timer.java:45]: Isn't it allargsconstructor?

File: src/main/java/com/team4/testingsystem/services/impl/TimerServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+@Service
+public class TimerServiceImpl implements TimerService {
+    private final TimerRepository timerRepository;
+    private final ApplicationEventPublisher eventPublisher;
+
+    @Autowired
+    public TimerServiceImpl(TimerRepository timerRepository,
+                            ApplicationEventPublisher eventPublisher) {
+        this.timerRepository = timerRepository;
+        this.eventPublisher = eventPublisher;
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/TimerServiceImpl.java]: Allargsconstructor

**Review by Anlord241 — APPROVED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

File: src/main/java/com/team4/testingsystem/entities/Timer.java (hunk 2 above)
Comment (maxim12321) [reply src/main/java/com/team4/testingsystem/entities/Timer.java:45]: Nope, field `id` isn't in the constructor

File: src/main/java/com/team4/testingsystem/services/impl/TimerServiceImpl.java (hunk 3 above)
Comment (maxim12321) [reply src/main/java/com/team4/testingsystem/services/impl/TimerServiceImpl.java]: Maybe we should then use lombok for all services?

File: src/main/java/com/team4/testingsystem/converters/notifications/TestExpiredNotificationConverter.java (hunk 1 above)
Comment (maxim12321) [reply src/main/java/com/team4/testingsystem/converters/notifications/TestExpiredNotificationConverter.java:20]: What means "the same"?
Is it about pushing `level` up to the abstract class?

Comment (AntonMaksimenkau) [reply src/main/java/com/team4/testingsystem/converters/notifications/TestExpiredNotificationConverter.java:20]: Yes

File: src/main/java/com/team4/testingsystem/services/impl/TimerServiceImpl.java (hunk 3 above)
Comment (AntonMaksimenkau) [reply src/main/java/com/team4/testingsystem/services/impl/TimerServiceImpl.java]: We have a task for it

File: src/main/java/com/team4/testingsystem/converters/notifications/TestExpiredNotificationConverter.java (hunk 1 above)
Comment (maxim12321) [reply src/main/java/com/team4/testingsystem/converters/notifications/TestExpiredNotificationConverter.java:20]: But `TestDeassignedNotificationConverter` doesn't set the level

**Pull request merged**